### DISCUSSION TITLE
Duplicate function: hasDatabaseConnection vs validateDatabaseConnection

### DISCUSSION THREAD

**Discussion by anlexN** [directus/directus#15470] [Archive]
![image](https://user-images.githubusercontent.com/8686739/189030975-b3617c34-c782-4328-90fc-eb5d767ef922.png)
I am reviewing the source code step by step, please explain why to split these functions for the same purpose. they are duplicates.

**Comment by anlexN** [comment]
@rijkvanzanten can you answer it?

**Reply by rijkvanzanten** [reply]
No need to tag me directly. I got notified for everything and will answer if and when I can

**Comment by rijkvanzanten** [comment]
There not the same. One returns a Boolean, the other will throw an error.
